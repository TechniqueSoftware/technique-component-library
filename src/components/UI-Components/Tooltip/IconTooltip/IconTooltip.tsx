import * as React from 'react';
import { Theme } from '@material-ui/core';
import Tooltip from '@material-ui/core/Tooltip';
import withStyles from '@material-ui/styles/withStyles';
import { colors } from '../../storyThemes';
import { SvgIconProps } from '@material-ui/core/SvgIcon/SvgIcon';

const styles = (theme: Theme) => ({
  icon: {
    fontSize: theme.typography.body2.fontSize,
    color: colors.icon.default(theme)
  }
});

export interface IconTooltipProps {
  tooltipText: string | React.ReactNode;
  placement?:
    | 'bottom-end'
    | 'bottom-start'
    | 'bottom'
    | 'left-end'
    | 'left-start'
    | 'left'
    | 'right-end'
    | 'right-start'
    | 'right'
    | 'top-end'
    | 'top-start'
    | 'top';
  iconProps?: SvgIconProps & { component: any };
  classes?: {
    icon?: string;
  };
}

export class IconTooltip extends React.PureComponent<IconTooltipProps> {

  static defaultProps: Partial<IconTooltipProps> = {
    classes: {}
  };

  render() {
    const {
      classes,
      placement = 'top',
      iconProps,
      tooltipText,
    } = this.props;

    const tooltipProps = {
      placement,
      title: tooltipText,
      tabIndex: 0,
    };

    const iconClassName = classes && classes.icon;
    const IconComponent = iconProps.component;

    return (
      <Tooltip {...tooltipProps}>
        <IconComponent classes={{ root: iconClassName }} {...iconProps} />
      </Tooltip>);
  }
}

export default withStyles(styles)(IconTooltip);
