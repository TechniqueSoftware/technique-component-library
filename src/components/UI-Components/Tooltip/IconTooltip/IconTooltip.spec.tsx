import * as React from 'react';
import { shallow } from 'enzyme';
import { IconTooltip, IconTooltipProps } from './IconTooltip';
import HelpRounded from '@material-ui/icons/HelpRounded';

describe('IconTooltip', () => {
  test('should render with minimal props', () => {

    const iconTooltipProps: IconTooltipProps = {
      tooltipText: 'mock text',
      iconProps: { component: HelpRounded },
      placement: 'bottom-start',
    };
    const shallowWrapper = shallow(<IconTooltip {...iconTooltipProps} />);

    expect(shallowWrapper.debug()).toMatchSnapshot();

  });
});
