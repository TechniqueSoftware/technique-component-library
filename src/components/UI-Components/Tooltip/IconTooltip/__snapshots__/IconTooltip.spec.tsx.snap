// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconTooltip should render with minimal props 1`] = `
"<WithStyles(ForwardRef(Tooltip)) placement=\\"bottom-start\\" title=\\"mock text\\" tabIndex={0}>
  <Memo(ForwardRef(HelpRoundedIcon)) classes={{...}} component={{...}} />
</WithStyles(ForwardRef(Tooltip))>"
`;
